File: JavaScript/leetcode_496.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function (nums1, nums2) {
    let next_greater = new Map();
    let stack = [];
    for (let index = 0; index < nums2.length; index++) {
        while (stack.length !== 0 && nums2[index] > stack.at(-1)) {
            let prev = stack.pop();
            next_greater.set(prev, nums2[index]);
        }
        stack.push(nums2[index]);
    }
    for (let index = 0; index < stack.length; index++) {
        next_greater.set(stack[index], -1);
    }
    return nums1.map((num) => next_greater.get(num));
};

console.log(nextGreaterElement((nums1 = [4, 1, 2]), (nums2 = [1, 3, 4, 2])));

File: Python/leetcode_496.py
from typing import List


def next_greater_element(nums1: List[int], nums2: List[int]) -> List[int]:
    result: List[int] = []
    for num in nums1:
        nums2_i = nums2.index(num)
        was_found = False
        for nums in nums2[nums2_i:]:
            if nums > num:
                result.append(nums)
                was_found = True
                break
        if was_found is False:
            result.append(-1)
    return result
